### DISCUSSION TITLE
Can't get any WiFi on my VM even with Spice Tools installed.

### DISCUSSION THREAD

**Discussion by StupidRepo** [utmapp/UTM#2416] [UTM Help (iOS)]
**Describe the issue**
Literally, getting WiFi is impossible, I have check EVERY setting yet it still doesn't work.

**Configuration**
* UTM Version: 25 (2.0.25)
* OS Version: Big Sur 11.1
* Intel or Apple Silicon? Apple Silicon ;)

**Additional Info**

I'm trying to install Cemu for the Wii U experience again ;)

**Comment by StupidRepo** [comment]
Wow, I got a crash when renaming my VM. I guess when I typed "pls kill me" as the name, it did it but on itself? I can't be bothered to upload the crash report since it doesn't matter to me. It's one of those U:tm: issues where it happens once and doesn't happen for a long long time.

~~*Relaunches U:tm:, re-renames VM, goes to QEMU tab, CRASH*~~ <- What's this doing here?

**Comment by conath** [comment]
You might be having trouble with the DNS inside your VMs, this is a common issue unfortunately. If you can, please try a DNS request on the command line / terminal in one of your VMs to see if that works.

Windows:

`nslookup duckduckgo.com 1.1.1.1`

Linux/macOS:

`dig duckduckgo.com @1.1.1.1`

If it returns an answer and does not have a timeout error, then your "WiFi" internet connection in the VM is working correctly and you need to change the DNS settings to make web browsers and other applications work, as well. Please search online for the name of the operating system you're using plus "manual DNS" to get instructions on that.

The DNS issue is already tracked as #2353.
